clab/clab.go

```diff
@@ -247,37 +262,58 @@ func (c *CLab) CreateLinks(ctx context.Context, workers uint, postdeploy bool) {
 	wg.Wait()
 }
 
-func (c *CLab) DeleteNodes(ctx context.Context, workers uint, containers []types.GenericContainer) {
+func (c *CLab) DeleteNodes(ctx context.Context, workers uint, deleteCandidates map[string]nodes.Node) {
 	wg := new(sync.WaitGroup)
 
-	ctrChan := make(chan *types.GenericContainer)
+	nodeChan := make(chan nodes.Node)
 	wg.Add(int(workers))
 	for i := uint(0); i < workers; i++ {
 		go func(i uint) {
 			defer wg.Done()
 			for {
 				select {
-				case cont := <-ctrChan:
-					if cont == nil {
+				case n := <-nodeChan:
+					if n == nil {
 						log.Debugf("Worker %d terminating...", i)
 						return
 					}
-					err := c.Runtime.DeleteContainer(ctx, cont)
+					err := n.Delete(ctx)
 					if err != nil {
-						log.Errorf("could not remove container '%s': %v", cont.ID, err)
+						log.Errorf("could not remove container %q: %v", n.GetName(), err)
 					}
 				case <-ctx.Done():
 					return
 				}
 			}
 		}(i)
 	}
-	for _, ctr := range containers {
-		ctr := ctr
-		ctrChan <- &ctr
+	for _, n := range deleteCandidates {
+		nodeChan <- n
 	}
-	close(ctrChan)
+	close(nodeChan)
 
 	wg.Wait()
 
 }
+
+func (c *CLab) ListContainers(ctx context.Context, labels []*types.GenericFilter) ([]types.GenericContainer, error) {
+	var containers []types.GenericContainer
+
+	for _, r := range c.Runtimes {
+		ctrs, err := r.ListContainers(ctx, labels)
+		if err != nil {
+			return containers, fmt.Errorf("could not list containers: %v", err)
+		}
+		containers = append(containers, ctrs...)
+	}
+	return containers, nil
+}
+
+func (c *CLab) GetNodeRuntime(query string) (runtime.ContainerRuntime, error) {
+	for _, node := range c.Nodes {
+		if node.GetName() == query {
+			return node.GetRuntime(), nil
+		}
+	}
+	return nil, fmt.Errorf("could not find a container matching name %q", query)
+}
```

Comment:
The problem here is that query is a `NodeConfig.LongName` but map's key is a `NodeConfig.ShortName`

Reply:
another option is to split the `LongName` by `labName` substring, this will produce a slice with two elements, where `[1]` is the element with the `ShortName`

Reply:
I've done that with a helper funciton.